test/lib/ansible_test/_internal/coverage/analyze/targets/generate.py

```diff
@@ -0,0 +1,143 @@
+"""Analyze code coverage data to determine which integration test targets provide coverage for each arc or line."""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+
+from .... import types as t
+
+from ....encoding import (
+    to_text,
+)
+
+from ....data import (
+    data_context,
+)
+
+from ....util_common import (
+    ResultType,
+)
+
+from ... import (
+    enumerate_powershell_lines,
+    enumerate_python_arcs,
+    get_collection_path_regexes,
+    get_powershell_coverage_files,
+    get_python_coverage_files,
+    get_python_modules,
+    initialize_coverage,
+    PathChecker,
+)
+
+from . import (
+    CoverageAnalyzeTargetsConfig,
+    get_target_index,
+    make_report,
+    write_report,
+)
+
+if t.TYPE_CHECKING:
+    from . import (
+        Arcs,
+        Lines,
+        TargetIndexes,
+    )
+
+
+def command_coverage_analyze_targets_generate(args):  # type: (CoverageAnalyzeTargetsGenerateConfig) -> None
+    """Analyze code coverage data to determine which integration test targets provide coverage for each arc or line."""
+    root = data_context().content.root
+    target_indexes = {}
+    arcs = dict((os.path.relpath(path, root), data) for path, data in analyze_python_coverage(args, target_indexes).items())
+    lines = dict((os.path.relpath(path, root), data) for path, data in analyze_powershell_coverage(args, target_indexes).items())
+    report = make_report(target_indexes, arcs, lines)
+    write_report(args, report, args.output_file)
+
+
+def analyze_python_coverage(
```

Comment:
This and the powershell functions are quite similar. Is there a way we could make that code generic which accepts a list of files and checks whether the coverage data is line or arc based. Would save on duplication and potentially make it easier to support other languages in the future.

Reply:
It should be possible to share more of the logic between the two. The same is probably true for the `coverage combine` implementations for Python and PowerShell. I'd like to save both of those for a follow-up PR though.